### PR TITLE
Split path refactor

### PR DESCRIPTION
**Pull request by LPeter1997**

### CODE REVIEW

File: src/Draco.Compiler/Internal/Declarations/DeclarationTable.cs [hunk 1]
```diff
@@ -90,17 +94,26 @@ private MergedModuleDeclaration BuildMergedRoot()
                     path, this.RootPath));
 
                 // Add to root so the compilation can continue
-                modules.Add(new SingleModuleDeclaration(rootName, rootNameSplitPath, (CompilationUnitSyntax)tree.Root));
+                modules.Add(new SingleModuleDeclaration(
+                    name: rootPath.Last,
+                    path: rootPath.Slice(^1..),
```
Comment (Binto86) [inline src/Draco.Compiler/Internal/Declarations/DeclarationTable.cs:99]: Why not use pathBeforeRoot?

Comment (Binto86) [reply src/Draco.Compiler/Internal/Declarations/DeclarationTable.cs:99]: here and in all the other places

File: src/Draco.Compiler/Internal/Declarations/DeclarationTable.cs [hunk 2]
```diff
@@ -90,17 +94,26 @@ private MergedModuleDeclaration BuildMergedRoot()
...
+                    syntax: (CompilationUnitSyntax)tree.Root));
                 continue;
             }
 
-            var subPath = path.RemovePrefix(this.compilation.SplitRootModulePath, true);
-            if (subPath.IsEmpty) subPath = rootNameSplitPath;
-            var name = subPath.Parts.Span[^1];
+            var subPath = path.RemovePrefix(pathBeforeRoot);
```
Comment (Binto86) [inline src/Draco.Compiler/Internal/Declarations/DeclarationTable.cs:104]: does this work? don't we need to have the last element or root here too?

File: src/Draco.Compiler/Internal/Declarations/DeclarationTable.cs (hunk 1 above)
Comment (LPeter1997) [reply src/Draco.Compiler/Internal/Declarations/DeclarationTable.cs:99]: `pathBeforeRoot` is everything before the root module. So if the root in `foo/bar/baz`, then `pathBeforeRoot` is `foo/bar` and `rootPath.Slice(^1..)` is `baz`

File: src/Draco.Compiler/Internal/Declarations/DeclarationTable.cs (hunk 2 above)
Comment (LPeter1997) [reply src/Draco.Compiler/Internal/Declarations/DeclarationTable.cs:104]: So my idea was that we strip off everything _before_ the root. Then every path from the root module works with paths relative to root

**Pull request merged**